astropy/stats/funcs.py

```diff
@@ -790,3 +793,101 @@ def mad_std(data):
 
     # NOTE: 1. / scipy.stats.norm.ppf(0.75) = 1.482602218505602
     return median_absolute_deviation(data) * 1.482602218505602
+
+
+# TODO Note scipy dependency
+def sigma_to_logp(sigma, two_sided=True):
+    """Convert number-of-sigma to log-probability
+
+    Detection probabilities are often naturally expressed as a
+    false positive probability. On the other hand, astronomers
+    are used to thinking of small probabilities in terms of
+    how many standard deviations out on a normal distribution
+    they are. This function converts between these two
+    representations in a way that works even for probabilities
+    too small to represent in floating point (this happens at
+    about 40-sigma for double precision).
+
+    Parameters
+    ----------
+    sigma : array-like
```

Comment:
Looks like this only works for arrays, not array-like as claimed here:
```
In [37]: sigma_to_logp([1, 2])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-37-96397e084aba> in <module>()
----> 1 sigma_to_logp([1, 2])

/Users/deil/code/astropy/astropy/stats/funcs.py in sigma_to_logp(sigma, two_sided)
    833     """
    834     import scipy.special
--> 835     r = scipy.special.log_ndtr(-sigma)+(np.log(2) if two_sided else 0)
    836     if two_sided:
    837         if np.any(sigma<0):

TypeError: bad operand type for unary -: 'list'
```
Add call to `np.asanyarray` at the start?